crates/chain/src/keychain/txout_index.rs

```diff
@@ -199,15 +174,10 @@ impl<K: Clone + Ord + Debug> KeychainTxOutIndex<K> {
 
     /// Set the lookahead count for `keychain`.
     ///
-    /// The lookahead is the number of scripts to cache ahead of the last stored script index. This
-    /// is useful during a scan via [`scan`] or [`scan_txout`].
-    ///
-    /// # Panics
-    ///
-    /// This will panic if the `keychain` does not exist.
-    ///
-    /// [`scan`]: Self::scan
-    /// [`scan_txout`]: Self::scan_txout
+    /// The lookahead is the number of scripts to cache ahead of the last revealed script index. This
+    /// is useful to find outputs you own when processing block data that lie beyond the last revealed
+    /// index. This is because in certain scenarios we might not know the last revealed indexit in like
```

Comment:
Thanks this explanation is better. There's a typo in this sentence. I think it just needs a small rewrite.

PS does this method still panic if the keychain doesn't exist? If so we should probably still mention that.

Reply:
Let me check. If it still panics, I will mention it.